doctr/models/recognition/core.py

```diff
@@ -93,8 +95,29 @@ def __call__(
             if any(crop.ndim != 3 for crop in crops):
                 raise ValueError("incorrect input shape: all crops are expected to be multi-channel 2D images.")
 
+            splitted_crops: List[np.ndarray] = []
+            splitted_idxs: List[List[int]] = []
+            for crop in crops:
+                h, w = crop.shape[:2]
```

Comment:
In case we have a channels_first tensor:
```python
channels_first = isinstance(crops[0], np.ndarray) or is_tf_available()
h, w = crop.shape[:2] if channels_first else crop.shape[-2:]
```